### PR TITLE
[HybridApp] Explanation modal

### PR DESCRIPTION
**Pull request by mateuuszzzzz**
PR added an Explanation Modal. We want to display it for first time `New Expensify` users and only in the `HybridApp`

### CODE REVIEW

File: src/libs/Navigation/AppNavigator/Navigators/ExplanationModalNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {createStackNavigator} from '@react-navigation/stack';
```
Comment (mateuuszzzzz) [inline src/libs/Navigation/AppNavigator/Navigators/ExplanationModalNavigator.tsx:1]: I added this modal as a separate navigator. To match the convention used for `WelcomeVideoModalNavigator`. Nevertheless, I think we can flatten the navigation structure here and move it to the root navigator. We keep there many screens that use `react-native-modal` under the hood. The same argument goes for `OnboardingWelcomeVideo.tsx`

File: src/components/ExplanationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import variables from '@styles/variables';
+import * as Welcome from '@userActions/Welcome';
+import ROUTES from '@src/ROUTES';
+import Text from './Text';
+import WelcomeVideoModal from './WelcomeVideoModal';
+
+function ExplanationModal() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [shouldNavigateToOnboarding, setShouldNavigateToOnboarding] = useState(false);
```
Comment (Skalakid) [inline src/components/ExplanationModal.tsx:14]: How about changing it to `shouldNavigateToOnboardingPage`?

File: src/components/TestToolMenu.tsx [hunk 3]
```diff
@@ -101,6 +105,23 @@ function TestToolMenu({user = USER_DEFAULT, network}: TestToolMenuProps) {
                     }}
                 />
             </TestToolRow>
+            <TestToolRow title="Explanation modal">
```
Comment (Skalakid) [inline src/components/TestToolMenu.tsx:108]: Maybe let's add small comment about it like the above TestToolRows have

**Review by Skalakid — COMMENTED** [review]
I left some small comments, but overall good job!

File: src/components/WelcomeVideoModal.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
+        if (isWelcomeVideoStatusLocked) {
+            return;
+        }
+
+        if (isOffline) {
+            setWelcomeVideoStatus('animation');
+        } else if (!isOffline) {
```
[the remark this answers is not in the document]
Comment (mateuuszzzzz) [reply src/components/WelcomeVideoModal.tsx:57]: Good point!

File: src/components/WelcomeVideoModal.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    children: React.ReactNode;
+    onClose?: () => void;
+};
+
+function WelcomeVideoModal({children, onClose}: WelcomeVideoModalProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [isModalVisible, setIsModalVisible] = useState(true);
+    const {shouldUseNarrowLayout} = useOnboardingLayout();
+    const [welcomeVideoStatus, setWelcomeVideoStatus] = useState<VideoStatus>('video');
```
[the remark this answers is not in the document]
Comment (mateuuszzzzz) [reply src/components/WelcomeVideoModal.tsx:44]: Totally agree, let's do it

**Review by Skalakid — APPROVED** [review]
LGTM 🦦

**Pull request closed without merging**